aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -619,6 +619,263 @@ Tensor matrix_rank(const Tensor& self, bool symmetric) {
   return at::linalg_matrix_rank(self, 0.0, c10::nullopt, symmetric);
 }
 
+std::tuple<Tensor&, Tensor&, Tensor&, Tensor&, Tensor&>
+_linalg_svd_rank_restricted_helper_out(
+    const Tensor& input,
+    const c10::optional<Tensor>& atol_opt,
+    const c10::optional<Tensor>& rtol_opt,
+    bool full_matrices,
+    bool compute_unique_rank,
+    Tensor& U,
+    Tensor& S,
+    Tensor& Vh,
+    Tensor& rank,
+    Tensor& unique_rank) {
+  at::linalg_svd_out(U, S, Vh, input, full_matrices);
+
+  // make sure rank is of proper size
+  const auto batch_shape = IntArrayRef(input.sizes().cbegin(), input.sizes().cend() - 2);
```

Comment:
aka `input.sizes().slice(0, input.dim() - 2);` but current variant is fine as well!

Reply:
I never remember that the `slice` method is a thing. I should start using it more.